apps/src/templates/teacherDashboard/LoginTypeParagraph.jsx

```diff
@@ -0,0 +1,124 @@
+import React, {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import i18n from '@cdo/locale';
+import {SectionLoginType} from '@cdo/apps/util/sharedConstants';
+import Button from '../Button';
+import ChangeLoginTypeDialog from './ChangeLoginTypeDialog';
+import {sectionShape} from './shapes';
+
+class LoginTypeParagraph extends Component {
+  static propTypes = {
+    sectionId: PropTypes.number.isRequired,
+    onLoginTypeChanged: PropTypes.func,
+    // Provided by Redux
+    section: sectionShape,
+  };
+
+  state = {
+    isDialogOpen: false,
+  };
+
+  openDialog = () => this.setState({isDialogOpen: true});
+  closeDialog = () => this.setState({isDialogOpen: false});
+  onLoginTypeChanged = () => {
+    this.closeDialog();
+    const callback = this.props.onLoginTypeChanged;
+    if (typeof callback === 'function') {
+      callback();
+    }
+  };
+
+  /**
+   * Pick content that varies with current login type and section student count.
+   * @param {SectionLoginType} loginType
+   * @param {number} studentCount
+   * @returns {{paragraph:ReactElement, buttonTest: string}}
+   */
+  static getContent(loginType, studentCount) {
+    let paragraph, buttonText;
+
+    if (loginType === SectionLoginType.picture) {
+      buttonText = i18n.changeLoginTypeToWord_button();
+      paragraph = (
+        <div>
+          <p>
+            {i18n.loginTypePictureLongDescription()}
+          </p>
+          <p>
+            {i18n.loginTypePictureResetDescription()}
+          </p>
+        </div>
+      );
+    } else if (loginType === SectionLoginType.word) {
+      buttonText = i18n.changeLoginTypeToPicture_button();
+      paragraph = (
+        <div>
+          <p>
+            {i18n.loginTypeWordLongDescription()}
+          </p>
+          <p>
+            {i18n.loginTypeWordResetDescription()}
+          </p>
+        </div>
+      );
+    } else if (loginType === SectionLoginType.email) {
+      buttonText = i18n.changeLoginTypeToWordOrPicture_button();
+      paragraph = (
+        <div>
+          <p>
+            {i18n.loginTypeEmailLongDescription()}
+          </p>
+          <p>
+            {i18n.loginTypeEmailResetDescription()}
+          </p>
+        </div>
+      );
+    }
```

Comment:
I also wonder whether getButtonText and getParagraph should be separate methods?